templates/mesh-gateway-deployment.yaml

```diff
@@ -160,11 +160,13 @@ spec:
                 service {
                   kind = "mesh-gateway"
                   name = "{{ default "mesh-gateway" .Values.meshGateway.consulServiceName }}"
+                  {{- if .Values.global.federation }}
```

Comment:
The joys of YAML templating 😃

Reply:
😆 yep. I was like "why isn't this short-circuiting?!!"

Reply:
😆 yep, I was like "why isn't this short circuiting!?!"